### PR TITLE
HRINT-2422 [Demo] Attachments: Index Attachment Details

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                }
+            ]
+        },
+        {
+            "Title": "Query for documents",
+            "Slug" : "step-5",
+            "DescriptionLines": [
+                "* Query the 'Employees' collection using the defined index.",
+                "",
+                "* The query results will include all documents from the 'Employees' collection that have attachments with a matching Type and Size."
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:139]: taking into account query is 'employee.AttachmentSizes.Any(x => x > attachmentSize)' would it be appropriate to state "minimum Size"?

Comment (Danielle9897) [reply DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json]: done

**Comment by Danielle9897** [comment]
@DejanMilicic Can you pls review ?

**Review by topalach — APPROVED** [review]

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json [hunk 2]
```diff
@@ -0,0 +1,160 @@
+{
+    "Slug": "index-attachment-details",
+    "SourceFileName": "IndexAttachmentDetailsController.cs",
+    "Title": "Index Attachment Details",
+    "DescriptionLines": [
+        "**attachment's details and content can both be indexed**.",
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:6]: First letter of this sentence is lower-case

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    ],
+
+    "Walkthroughs": [
+        {
+            "Title": "Define the index class",
+            "Slug": "step-1",
+            "DescriptionLines": [
+                "* This example defines a Map index on the _Employees_ collection."
+            ],
+            "Assets": [
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:46]: image

Do you think that small padding below the line would improve looks?

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        },
+        {
+            "Title": "Load details for indexing",
+            "Slug": "step-3",
+            "DescriptionLines": [
+                "* In the index Map function, call `AttachmentsFor()` to load the document's attachment details to be indexed.",
+                "",
+                "* The _attachments_ array will hold the following per attachment:",
+                "  * Name",
+                "  * Type",
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:88]: `Content Type` might be slightly more precise

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            ]
+        },
+        {
+            "Title": "Configure index fields",
+            "Slug" : "step-4",
+            "DescriptionLines": [
+                "* Configure the index-fields content.",
+                "",
+                "* The attachments names, types, hashes & sizes are now available and can be used to configure the index fields.<br>",
+                "  e.g., _AttachmentNames_ is an index field that holds all the attachments names of the document.",
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:119]: "holds all the attachments names of the document" -> "holds all the attachments' names for the document"?

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                "",
+                "* Later on, the 'Employees' collection can be queried using this index to get employees that have specific attachments names, types, etc.",
+                "",
+                "* Re-indexing will happen on any change to the document's attachments."
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:123]: So, indexing is smart enough to detect that only attachments are taken from the documents and changes to the document's JSON will not trigger reindexing for this index?

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+                }
+            ]
+        },
+        {
+            "Title": "Query for documents",
+            "Slug" : "step-5",
+            "DescriptionLines": [
+                "* Query the 'Employees' collection using the defined index.",
+                "",
+                "* The query results will include all documents from the 'Employees' collection that have attachments with a matching Type and minimum Size."
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:139]: with a matching Content Type might be a slightly more precise formulation?

**Review by DejanMilicic — APPROVED** [review]

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json (hunk 3 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:46]: There is this issue I opened 2 weeks ago about that
https://issues.hibernatingrhinos.com/issue/HRINT-2468

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json (hunk 2 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:6]: done

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json (hunk 4 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:88]: done

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json (hunk 7 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:139]: done

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json (hunk 5 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:119]: done (added apostrophe)
but prefer to leave 'of the document'

File: DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json (hunk 6 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/Attachments/IndexAttachmentDetails/metadata.json:123]: yes - as far as I understand.
only changes made to the attachments will trigger indexing.
(that does include deleting a document that had attachments..)

**Review by DejanMilicic — APPROVED** [review]

**Pull request merged**